File: backend/app/models/finding.py
# finding.py
from pydantic import BaseModel
from typing import List


class Finding(BaseModel):

    id: int

    finding_id: str

    title: str

    severity: str

    risk_score: int

    category: str

    description: str

    impact: str

    evidence: str

    remediation: str

    references: List[str]

    compliance: List[str]

    tags: List[str]

    server: str

    host: str

    status: str

    auto_fix_supported: bool = False

    requires_restart: bool = False

    requires_reboot: bool = False

    manual_review: bool = False

File: frontend/src/types/finding.ts
// Findings.ts
export interface Finding {

  id: string

  asset_id: number

  finding_id: string

  title: string

  severity: string

  risk_score: number

  category: string

  description: string

  impact: string

  evidence: string

  remediation: string

  references: string[]

  compliance: string[]

  tags: string[]

  server: string

  host: string

  ip_address: string

  status: string

  auto_fix_supported: boolean

  requires_restart: boolean

  requires_reboot: boolean

  manual_review: boolean
}